controllers/mirrorpeersecret_controller.go

```diff
@@ -97,5 +102,34 @@ func mirrorPeerSecretReconcile(ctx context.Context, rc client.Client, req ctrl.R
 func (r *MirrorPeerSecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Secret{}, builder.WithPredicates(utils.SourceOrDestinationPredicate)).
+		Watches(&source.Kind{Type: &corev1.ConfigMap{}}, handler.EnqueueRequestsFromMapFunc(r.secretConfigMapFunc)).
 		Complete(r)
 }
+
+func (r *MirrorPeerSecretReconciler) secretConfigMapFunc(obj client.Object) []reconcile.Request {
+	ConfigMapRamenConfigKeyName := "ramen_manager_config.yaml"
+	var cm *corev1.ConfigMap
+	cm, ok := obj.(*corev1.ConfigMap)
+	if !ok {
+		return []reconcile.Request{}
+	}
+
+	ramenConfig := &ramendrv1alpha1.RamenConfig{}
+	err := yaml.Unmarshal([]byte(cm.Data[ConfigMapRamenConfigKeyName]), ramenConfig)
+	if err != nil {
+		return []reconcile.Request{}
+	}
+
+	secrets := &corev1.SecretList{}
+	if err := r.Client.List(context.TODO(), secrets); err != nil {
+		return []reconcile.Request{}
+	}
+
+	requests := make([]reconcile.Request, len(secrets.Items))
+	for i, secret := range secrets.Items {
+		requests[i].Name = secret.GetName()
+		requests[i].Namespace = secret.GetNamespace()
+	}
```

Comment:
Do we need to requeue for all the secrets or can we filter it down based on labels/annotations?

Reply:
I've added label selector to just select the internal secrets (s3 types)